runtime_scan/pkg/provider/gcp/client_test.go

```diff
@@ -0,0 +1,168 @@
+package gcp
+
+import (
+	"reflect"
+	"testing"
+
+	"cloud.google.com/go/compute/apiv1/computepb"
+
+	"github.com/openclarity/vmclarity/api/models"
+)
+
+func Test_convertTags(t *testing.T) {
+	type args struct {
+		tags *computepb.Tags
+	}
+	tests := []struct {
+		name string
+		args args
+		want *[]models.Tag
+	}{
+		{
+			name: "sanity",
+			args: args{
+				tags: &computepb.Tags{
+					Items: []string{"tag1", "tag2=val2"},
+				},
+			},
+			want: &[]models.Tag{
+				{
+					Key:   "tag1",
+					Value: "",
+				},
+				{
+					Key:   "tag2",
+					Value: "val2",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := convertTags(tt.args.tags); !reflect.DeepEqual(got, tt.want) {
```

Comment:
I would use `gomega` as a testing library as it is more sophisticated and support comparing types where `reflect.DeepEqual` would fail. I'd move away from using `DeepEqual` for testing for sure. Not to mention that it also makes tests more readable.
Example for using `gomega`: https://github.com/openclarity/vmclarity/blob/main/runtime_scan/pkg/provider/cloudinit/cloudinit_test.go

Reply:
alternatively the standard we use across the rest of the project is cmp.Equals https://github.com/google/go-cmp it must nicer and doesn't pull in a ton of other dependencies

Reply:
`gomega` also relies on `go-cmp` in certain scenarios, but it provides much better DX in my opinion as it also includes matchers for other use-cases where neither `go-cmp` nor `DeepEqual` is appropriate.
See the docs: https://onsi.github.io/gomega/#provided-matchers

> it must nicer and doesn't pull in a ton of other dependencies

I'd not say that it does pull in a ton of deps: https://github.com/onsi/gomega/blob/master/go.mod
Actually it is quite slim compared to average go projects.

Anyways, we need to decide what are the tools we want to use for testing. Personally, I find both `go-cmp` and `reflect.DeppEqual` limited, so I do prefer using `gomega` for unit tests.

Reply:
gomega looks very nice but for my use cases I think I can just stick with reflect.DeepEqual. 
Anyway this is what gomega.equal is doing